docs/synapse/userguides/storm_ref_automation.rstorm

```diff
@@ -12,72 +12,136 @@ Storm Reference - Automation
-
-**Review and Testing.**
-
-Automation in Synapse provides significant advantages. It relieves analysts from performing tedious work, freeing them to focus on more detailed analysis and complex tasks. It also allows you to scale your analytical operations by limiting the amount of work that must be performed manually. However, badly written automation or automation that contains logical errors also allows you to "make errors at machine speed". We strongly encourage testing automation in a development environment before implementing it on a production system.
-
-**Nodes In and Nodes Out.**
-
-In some cases automation can be used to "chain" operations (e.g., both triggers and cron can call macros) or may be executed inline with other Storm operations (a macro can be called in the middle of a Storm query where the query expects to perform further operations on the nodes returned by the macro). In these cases, both the automation itself and any Storm executed after the automation may fail if the inbound nodes are not what is expected by the query.
+- Individual triggers and cron jobs can be kept entirely separate from one another, each executing
+  their own dedicated Storm code. This approach helps keep automation "self-contained" and means the
+  Storm executed by a given trigger or cron job is directly introspectable via Storm itself (as a
+  property of ``syn:trigger`` or ``syn:cron`` nodes). However, it may provide less flexibility in
+  executing the associated Storm compared with the use of macros.
+  
+  Alternatively, tasks such as triggers and cron jobs can be written to execute minimal Storm queries
+  whose purpose is to call more extensive Storm stored centrally in macros. This approach consolidates
+  much of the associated Storm, which may make it easier to manage and maintain. Storm placed in macros
+  also provides flexibility as the macro can be called by a trigger, a cron job, or a user as part of
+  a Storm query.
+
+- Automation can be written as many small, individual elements. Each element can perform a relatively
+  simple task, but the elements can work together like building blocks to orchestrate larger-scale
+  operations. This approach keeps tasks "bite sized" and the Storm executed by a given piece of automation
+  generally simpler. However it may result in a larger number of automation elements to maintain and
+  make it more challenging to understand the potential interactions of so many different elements.
+  
+  Alternatively, automation can be implemented using fewer elements that perform larger, more unified
+  tasks (or that consolidate numerous smaller tasks). This approach results in fewer automation elements
+  overall, but typically requires you to write and maintain more advanced Storm (e.g., to create a small
+  number of macros with switch or if/else statements to each manage a variety of tasks). However, the
+  Storm is consolidated in fewer locations, which may make managing and troubleshooting easier.
```

Comment:
This is very minor but "consolidated in fewer locations" reads as slightly redundant - maybe "consolidated within a few locations"?